src/Facebook/FacebookBatchRequest.php

```diff
@@ -189,8 +191,19 @@ public function convertRequestsToJson()
     {
         $requests = [];
         foreach ($this->requests as $request) {
-            $attachedFiles = isset($request['attached_files']) ? $request['attached_files'] : null;
-            $requests[] = $this->requestEntityToBatchArray($request['request'], $request['name'], $attachedFiles);
+            $options = [];
+
+            if (!empty($request['name'])) {
+                $options['name'] = $request['name'];
+            }
+
+            if (!empty($request['attached_files'])) {
```

Comment:
@yguedidi Then why not just use the `isset()` function that returns false when the variable is null. Plus I think that this is how we check in similar cases throughout the project.